visivo/server/project_writer.py

```diff
@@ -1,32 +1,183 @@
 import json
+import os
+import ruamel.yaml 
 from copy import deepcopy
 
-class ProjectNamedChildrenWriter:
-    def __init__(self, named_children, project_file_path):
+DELETE = object()
+
+class SubDiff(dict):
+    pass
+
+def diff(old_dict, new_dict):
+    updates = {}
+    for key in new_dict:
```

Comment:
I think for these to be maintainable, it is probably best to break out these logically pieces into private methods with a name.  I think other than that it is going to be hard to track down bugs in these code. Then it can read more like:

```
for key in new_dict:
  __record_updates_not_in_old_dict(key, new_dict)
  __record_updates...
  ...
```